query-engine/prisma-models/src/selection_result.rs

```diff
@@ -0,0 +1,172 @@
+use crate::{DomainError, FieldSelection, PrismaValue, ScalarFieldRef, SelectedField};
+use std::convert::TryFrom;
+
+/// Represents a set of results.
+#[derive(Default, Debug, Clone, PartialEq, Eq, Hash)]
+// Name candidates:
+// - ReturnValues
+// - PrismaValues
+// - Values
+// - ResultValues
+// - SelectionResult << This would remove SelectedField and reuse the SelectedField concept.
+//
+// My hope: Generalizing the return value container will allow us to move a lot of
+// hacks into the normal return value stream instead of handling that differently.
+pub struct SelectionResult {
+    pub pairs: Vec<(SelectedField, PrismaValue)>,
```

Comment:
Why not use a BtreeMap or HashMap here?

Reply:
`HashMap` can't be used because it's not `Hash` or `Ord` for example, which is required for us to make certain operations and comparators work in the core. You can of course provide your own, but it's a lot of implementation for nothing.

`BTreeMap` does implement both the required traits - great! But it doesn't really do much more here than a vector as the rust BTreeMap is linear search as well. On the upside for vector, you again get a fixed ordering of items that you can leverage if needed, while a BTreeMap orders based on the key or value ordering, not sure which was which.